cdragontoolbox/wad.py

```diff
@@ -222,6 +222,17 @@ def set_unknown_paths(self, path):
                 else:
                     wadfile.path = f"{path}/{wadfile.path_hash:016x}"
 
+    def sanitize_paths(self):
+        """Truncate files that have a length of over 255"""
```

Comment:
I think you should also truncate basename whose length is exactly 255.
This way, all files whose basename is 255 use the "truncated format". This could be use to detect truncated paths.
I'm not sure this will be used one day, but it costs nothing. :)

Reply:
Also not sure whether that should be done, but if you think it's better that way, I'll add it.